Updater changes needed by TAF

renatav opened the pull request:
Changes needed to support development of our updater.

renatav committed:
Made check if metadata is expired more flexible

Made it possible to specify the earliest valid expiration data of a metadata file. This can be redefined using the update handler.

renatav committed:
Modifications of the updater needed to implement the TAF updater

Moved downloading a target file and cheking its digest to the handle

renatav committed:
Added support for checking if metadata file remained the same

This can be checked when refreshing metadata, to ensure that all metadata files which should've remained the same, did remain the same

renatav committed:
Renamed wrongly named parameter

renatav committed:
Fixes needed to prevent failing build

- Added a pylint disable in two places
- Removed an unused import

danixeee commented on tuf/client/updater.py:
```diff
@@ -2334,14 +2330,15 @@ def _ensure_not_expired(self, metadata_object, metadata_rolename):
     # with microseconds attached.)
-    current_time = int(time.time())
+
+    time = self.update_handler.earliest_valid_expiration_time(metadata_rolename)
```
`Redefining name 'time' from outer scope (line 127)pylint(redefined-outer-name)`

Unused import `time` on line 127, but I would rename this variable to `earliest_exp_time` or some similar, more meaningful, name.

danixeee commented on tuf/client/handlers.py:
```diff
@@ -35,7 +37,15 @@ def __init__(self, mirrors, repository_directory):
     self.mirrors = mirrors
     self.repository_directory = repository_directory
 
+  def earliest_valid_expiration_time(self, metadata_rolename):
```
If `metadata_rolename` is not always needed,I would set default `None` value

danixeee commented on tuf/client/handlers.py:
```diff
@@ -35,7 +37,15 @@ def __init__(self, mirrors, repository_directory):
+    return int(time.time())
 
+  def ensure_not_changed(self, metadata_filename):
+    # not necessary in all cases
```
docstring will be helpful here

renatav committed:
Using kwargs in the handlers

We do not always need the same date in our git handler as
the original tuf implementation needs. By using kwargs, we
can avoid having to add pylint disable. This was necessary
in order to get the build passing. Removed long docstrings as
they just made the code less readable and this is never going
to get merged upstream in any case

danixeee commented on tuf/client/handlers.py:
```diff
@@ -92,64 +99,26 @@ class RemoteMetadataUpdater(MetadataUpdater):
-      self.mirrors)
-
+  def get_mirrors(self, **kwargs):
+    file_type = kwargs['file_type']
```
`kwargs.pop('file_type', None)` to avoid `KeyError`. Same for line 104.

danixeee commented on tuf/client/handlers.py:
```diff
@@ -92,64 +99,26 @@ class RemoteMetadataUpdater(MetadataUpdater):
-      contents of 'url'.
-    """
+  def get_metadata_file(self, file_mirror, **kwargs):
+    upperbound_filelength = kwargs['upperbound_filelength']
```
`kwargs.pop('upperbound_filelength', None)`

danixeee commented on tuf/client/handlers.py:
```diff
@@ -92,64 +99,26 @@ class RemoteMetadataUpdater(MetadataUpdater):
+        upperbound_filelength)
+
+  def get_target_file(self, file_mirror, **kwargs):
+    download_safely = kwargs['download_safely']
```
`kwargs.pop('download_safely', True)` if not passed? Line 115. should also use `pop`

renatav replied:
Resolved all

renatav committed:
Passing repo name to handler, removed kwargs

renatav committed:
Fixed get_target_file

renatav committed:
Added docstring

renatav committed:
Renamed variable, as per CR comment

renatav committed:
No longer requireing our oll-securesystemslib fork

danixeee reviewed (approved):
LGTM; merging
